### PR TITLE
Feature/user class and tests

### LINKED ISSUE #9: Build a User Class
Properties:
- [x] name
- [x] id
- [x] pantry (new instance of Pantry Class)
- [x] favoriteRecipes (array)
- [x] recipesToCook (array)
Methods:
- [x] `addFavoriteRecipe()` - takes in a recipe, pushes to `favoriteRecipes` array property,
check for duplicates
- [x] `removeFavoriteRecipe()` - remove recipe from `favoriteRecipes` array property
- [x] `decideToCook()` - takes in a recipe, pushes to `recipesToCook` array property 
- [x] `filterFavoriteRecipesTag()` - return list of recipes filtered by tag(s), from `favoriteRecipes` array
- [x] `filterFavoriteRecipesName()` - return list of recipes filtered by a recipe name, from `favoriteRecipes` array
- [x] - [ ] `filterFavoriteRecipesIngredient()` - return list of recipes filtered by a recipe ingredient, from `favoriteRecipes` array

Test:
- [x] All class properties
- [x] All methods and updates to properties
- [x] Any helper methods you build out
- [x] Happy/sad path tests
- [x] Create own data to test

### PR DESCRIPTION
**Pull request by elizhahn**
## Is this a fix or a feature?
- feature

## What is the change?
- this adds a testing suite file and class file for the User class
- User Stories: 
- As a user, I should be able to favorite / unfavorite recipes that I like and can easily find again.
- As a user, I should be able to filter my favorited recipes by one or more tags.
- As a user, I should be able to search my favorited recipes by its name or ingredients.
- As a user, I should be able to add a recipe to a list of recipes to cook.

## What does it fix?
- N/A
## Where should the reviewer start?
- user.js file
- user-test.js file

## How should this be tested?
- run npm test and all tests should be passing

### CODE REVIEW

File: src/user.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    return filteredRecipes; 
+  }
+
+  filterFavoritesByName(recipeName) {
+    const searchRecipeName = recipeName.toLowerCase();
+    return this.favoriteRecipes.find(recipe => recipe.name.toLowerCase() === searchRecipeName);
+  }
+}
+
+module.exports = User;
```
Comment (alexthompson207) [inline src/user.js]: Should this be 
`if (typeof module !== 'undefined') {
  module.exports = User;
}`. ?

**Review by alexthompson207 — COMMENTED** [review]
Great work on these methods, and happy/sad test paths!

File: src/user.js (hunk 1 above)
Comment (elizhahn) [reply src/user.js]: yes it should be!

**Comment by alexthompson207** [comment]
Looks good! Great job with all the tests and methods. 
We need to remember to go back later and refactor test files so that they aren't using `require`

**Pull request merged**